api/src/routes/settings.ts

```diff
@@ -163,6 +163,55 @@ export const settingRoutes: FastifyPluginCallbackTypebox = (
     }
   );
 
+  fastify.put(
+    '/update-my-about',
+    {
+      schema: {
+        body: Type.Object({
+          about: Type.Optional(Type.String()),
+          name: Type.Optional(Type.String()),
+          picture: Type.Optional(Type.String()),
+          location: Type.Optional(Type.String())
+        }),
+        response: {
+          200: Type.Object({
+            message: Type.Literal('flash.updated-about-me'),
+            type: Type.Literal('success')
+          }),
+          500: Type.Object({
+            message: Type.Literal('flash.wrong-updating'),
+            type: Type.Literal('danger')
+          })
+        }
+      }
+    },
+    async (req, reply) => {
+      let hasProtocol = false;
+      if (req.body.picture)
+        hasProtocol = Boolean(new URL(req.body.picture).protocol);
```

Comment:
```suggestion
      let hasProtocol = true;
      try {
        const url = req.body.picture ? new URL(req.body.picture) : null;
        hasProtocol =
          !!url && (url.protocol == 'http:' || url.protocol == 'https:');
      } catch {
        hasProtocol = false;
      }
```
This should work, but I recommend creating a little helper function and testing that separately. That means we'll have good coverage without having to have tests for lots of different urls.